src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -2285,112 +2285,190 @@ LogicalResult ONNXPadOp::inferShapes(
 // Unsqueeze
 //===----------------------------------------------------------------------===//
 
-LogicalResult ONNXUnsqueezeOp::inferShapes(
-    std::function<void(mlir::Region &)> doShapeInference) {
-  return emitError("Not implemented");
+// Update axes attribute so that it contains only positive values.
+// Helper functions for both Unsqueeze and Squeeze Ops
+template <typename Op>
+void updateNegativeAxis(Op *op, ArrayRef<int64_t> axes) {
+  OpBuilder builder(op->getContext());
+  if (auto axesConstOp = getONNXConstantOp(op->axes())) {
+    auto tensorType = axesConstOp.getType().template cast<RankedTensorType>();
```

Comment:
What's the difference in `template cast` and just `cast`? Should we change all the usage?

Reply:
`template cast` is needed here to resolve a compiler error because `updateNegativeAxis` is templatized.